### PR TITLE
Analytics: Add API to get submission count for a particular challenge.

### PR DESCRIPTION
**Pull request by taranjeet**
@deshraj @RishabhJain2018

### CODE REVIEW

File: apps/analytics/views.py [hunk 1]
```diff
@@ -42,3 +50,41 @@ def get_participant_count(request, challenge_pk):
...
+@permission_classes((permissions.IsAuthenticated, HasVerifiedEmail, IsChallengeCreator))
+@authentication_classes((ExpiringTokenAuthentication,))
+def get_submission_count(request, challenge_pk, duration):
+    '''
+        Returns submission count for a challenge according to the duration
+        Valid values for duration are all, daily, weekly and monthly.
+    '''
+    # make sure that a valid url is requested.
+    if duration.lower() not in ('all', 'daily', 'weekly', 'monthly'):
+        response_data = {'error': 'Wrong url pattern!'}
```
Comment (deshraj) [inline apps/analytics/views.py]: Please change the message to `Wrong URL pattern!`

File: apps/analytics/views.py [hunk 2]
```diff
@@ -42,3 +50,41 @@ def get_participant_count(request, challenge_pk):
...
+
+    challenge_phase_ids = challenge.challengephase_set.all().values_list('id', flat=True)
+
+    q_params = {'challenge_phase__id__in': challenge_phase_ids}
+    since_date = None
+    if duration.lower() == 'daily':
+        since_date = timezone.now().date()
+
+    elif duration.lower() == 'weekly':
+        since_date = (timezone.now() - timedelta(days=1)).date()
```
Comment (deshraj) [inline apps/analytics/views.py]: Shouldn't the timedelta should be `7` instead of `1`? Please let me know if I am wrong.

Comment (taranjeet) [inline apps/analytics/views.py]: yeah sure, just missed it.

File: apps/analytics/views.py (hunk 1 above)
Comment (taranjeet) [inline apps/analytics/views.py]: Yeah, I will update that.

**Comment by taranjeet** [comment]
@deshraj : I have updated the changes.

**Pull request merged**